cpp/src/gandiva/precompiled/time.cc

```diff
@@ -828,4 +828,44 @@ gdv_int64 castBIGINT_daytimeinterval(gdv_day_time_interval in) {
          extractDay_daytimeinterval(in) * MILLIS_IN_DAY;
 }
 
+#define CAST_INT_YEAR_INTERVAL(NAME, OUT_TYPE)                \
+  FORCE_INLINE                                                \
+  gdv_##OUT_TYPE NAME##_year_interval(gdv_year_interval in) { \
+    return static_cast<gdv_##OUT_TYPE>(in / 12.0);            \
```

Comment:
Because the year interval is represented as a month interval(Arrow only supports month intervals and daytime intervals). So I changed the type alias to gdv_month_interval to make it clear.